new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItConfigDistributionStrategy.java

```diff
@@ -0,0 +1,1056 @@
+    createJdbcDataSource("root", "root123", mysqlDBPort1);
+    //deploy application to view server configuration
+    deployApplication();
+
+  }
+
+  /**
+   * Verify the default config before starting any test.
+   */
+  @BeforeEach
+  public void beforeEach() {
+
+    String maxMessageSize = "10000000";
+    String cpMaxCapacity = "15";
+    String dsUrl = dsUrl1;
+    verifyConfig(maxMessageSize, cpMaxCapacity, dsUrl, false);
+
+  }
+
+  /**
+   * Delete the overrides, secrets and restart domain to get clean state.
+   */
+  @AfterEach
+  public void afterEach() {
+    deleteConfigMap(overridecm, domainNamespace);
+    deleteSecret(dsSecret, domainNamespace);
+    String patchStr
+        = "["
+        + "{\"op\": \"remove\", \"path\": \"/spec/configuration/overridesConfigMap\"},"
+        + "{\"op\": \"remove\", \"path\": \"/spec/configuration/secrets\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}", patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH);
+    restartDomain();
+  }
+
+  /**
+   * Test server configuration and JDBC datasource configurations are overridden dynamically when
+   * /spec/configuration/overrideDistributionStrategy: field is not set. By default it should be DYNAMIC.
+   *
+   * <p>Test sets the /spec/configuration/overridesConfigMap and /spec/configuration/secrets with new configuration
+   * and new secrets.
+   *
+   * <p>Verifies after introspector runs the server configuration and JDBC datasource configurations are updated
+   * as expected.
+   */
+  @Order(1)
+  @Test
+  @DisplayName("Test overrideDistributionStrategy set to DEFAULT")
+  public void testDefaultOverride() {
+
+    //store the pod creation timestamps
+    storePodCreationTimestamps();
+
+    //create config override map and secrets
+    setupCustomConfigOverrides();
+
+    //patch the domain resource with overridesConfigMap, secrets and introspectVersion
+    String patchStr
+        = "["
+        + "{\"op\": \"add\", \"path\": \"/spec/configuration/overridesConfigMap\", \"value\": \"" + overridecm + "\"},"
+        + "{\"op\": \"add\", \"path\": \"/spec/configuration/secrets\", \"value\": [\"" + dsSecret + "\"]  },"
+        + "{\"op\": \"add\", \"path\": \"/spec/introspectVersion\", \"value\": \"1\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}", patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    verifyIntrospectorRuns();
+    verifyPodsStateNotChanged();
+
+    //print the configuration overrides without asserting
+    verifyConfig(null, null, null, true);
+
+    //workaround for bug - setting overridesConfigMap doesn't apply overrides dynamically, needs restart of server pods
```

Comment:
This is not a bug. The overrides will only be applied following introspection. We are clarifying the rules about when introspection will happen. Russ has pending PR https://github.com/oracle/weblogic-kubernetes-operator/pull/1746 where we've clarified the rules as follows:

On restartVersion change, all affected pods are restarted in all modes
On introspectVersion change, introspection is redone in all modes. For model-in-image, if the introspection generates a new domain home then any running servers will be rolled to the new domain home.
Introspection will also occur if there is no current introspection-results config map or if the domain presently has zero server instances running and is now starting a server.
For model-in-image only, introspection will also run for every ‘make-right’ flow in which it is determined that a server pod must be updated. In particular, that entails rerunning from the start of the step chain, and this time running the introspector.
If introspection fails, the domain resource status must reflect the failure; however, the ‘make-right’ flow otherwise stops and any currently running pods are left intact.

Reply:
I still see it is not getting overridden even after waiting for significant amount of time.

Reply:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/548/console